### PR TITLE
API to get media attachment

### PR DESCRIPTION
**Pull request by Tunous**
Needed to determine whether uploaded media has finished processing if it didn’t upload immediately.

https://docs.joinmastodon.org/methods/media/#get

### CODE REVIEW

**Review by kkostov — CHANGES_REQUESTED** [review]
Looks great! I suggested a few minor tweaks to better reflect how the method will work between different flavours of the API.

File: Sources/TootSDK/TootClient/TootClient+Media.swift [hunk 1]
```diff
@@ -55,4 +55,24 @@ public extension TootClient {
         }
         return try await fetch(MediaAttachment.self, req)
     }
+    
+    /// Get a media attachment, before it is attached to a status and posted, but after it is accepted for processing.
+    /// Use this method to check that the full-sized media has finished processing.
+    ///
+    /// - Parameter id: The ID of the attachment.
+    /// - Returns: `Attachment` if media file was processed, and a `url` to processed media is available. `nil` otherwise.
```
[the remark this answers is not in the document]
Comment (Tunous) [reply Sources/TootSDK/TootClient/TootClient+Media.swift:63]: For future, is there a documentation about how this works on Pleroma? I tried to look for that endpont on their [api documentation](https://api.pleroma.social/) but didn’t find any mentions. That’s why I based implementation only on Mastodon.

[the remark this answers is not in the document]
Comment (kkostov) [reply Sources/TootSDK/TootClient/TootClient+Media.swift]: Good question!
I use several sources:

* There is some documentation on their website on how Pleroma differs from Mastodon https://docs-develop.pleroma.social/backend/development/API/differences_in_mastoapi_responses/#limitations 
* There is their official open api spec https://api.pleroma.social 
* I check the source code as a last resort https://git.pleroma.social/pleroma/pleroma/-/blob/develop/lib/pleroma/web/api_spec/operations/media_operation.ex 
* I create an example in `swiftyadmin` to confirm the behaviour on my instance. For example, in this case I would create a command like:

```swift
// /Sources/Media/GetMedia.swift

import ArgumentParser
import Foundation
import TootSDK

struct GetMedia: AsyncParsableCommand {

  @Option(name: .short, help: "URL to the instance to connect to")
  var url: String

  @Option(name: .short, help: "Access token for an account with sufficient permissions.")
  var token: String

  @Option(
    name: .short,
    help:
      "id of the media to retrieve"
  )
  var id: String

  mutating func run() async throws {
    print("Getting media with id: \(id)")
    let client = TootClient(instanceURL: URL(string: url)!, accessToken: token)

    let attachment = try await client.getMedia(id: id)
    let json = String.init(data: try TootEncoder().encode(attachment), encoding: .utf8)
    print(json ?? "")
  }
}
```

**Comment by kkostov** [comment]
Good question @Tunous!
I use several sources:

* There is some documentation on their website on how Pleroma differs from Mastodon https://docs-develop.pleroma.social/backend/development/API/differences_in_mastoapi_responses/#limitations 
* Pleroma's open api spec https://api.pleroma.social 
* I check the source code as a last resort https://git.pleroma.social/pleroma/pleroma/-/blob/develop/lib/pleroma/web/api_spec/operations/media_operation.ex 
* I create an example in `swiftyadmin` to confirm the behaviour on my instance. For example, in this case I would create a command like:

```swift
// /Sources/Media/GetMedia.swift

import ArgumentParser
import Foundation
import TootSDK

struct GetMedia: AsyncParsableCommand {

  @Option(name: .short, help: "URL to the instance to connect to")
  var url: String

  @Option(name: .short, help: "Access token for an account with sufficient permissions.")
  var token: String

  @Option(
    name: .short,
    help:
      "id of the media to retrieve"
  )
  var id: String

  mutating func run() async throws {
    print("Getting media with id: \(id)")
    let client = TootClient(instanceURL: URL(string: url)!, accessToken: token)

    let attachment = try await client.getMedia(id: id)
    let json = String.init(data: try TootEncoder().encode(attachment), encoding: .utf8)
    print(json ?? "")
  }
}
```

**Review by kkostov — APPROVED** [review]

**Pull request merged**